contents/docs/plugins/build.md

```diff
@@ -124,51 +189,60 @@ async function processEvent(event, { config, cache }) {
+Here's the same example from above, except now using `processEventBatch`: 
 
-#### cache
-
-A way to store values that persist across `processEvent` calls. The values are stored in [Redis](https://redis.io/), an in-memory store.
+```js
+async function processEventBatch(events, { config, cache }) {
+    const counter = await cache.get('counter', 0)
+    cache.set('counter', counter + 1)
 
-Storing values is done via `cache.set`, which takes the following parameters:
+    for (let event of events) {
+      if (event.properties) {
+          event.properties['hello'] = 'world'
+          event.properties['bar'] = config.bar
+          event.properties['$counter'] = counter
+      }
+    }
 
-```
-key: string, 
-value: unknown
+    return events
+}
 ```
 
-And retrieving them uses `cache.get`, which takes:
+As you can see, since `events` is an array of events, we iterate over it to access every individual event. 
 
-```
-key: string, 
-defaultValue: unknown
-```
+> **Note:** As of right now, batch size is not yet configurable and still receives only one event (albeit in array form).
 
-> If you're unfamiliar with TypeScript, you can read about the `unknown` type on this [blog post by Microsoft](https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type)
+### Scheduled Tasks
 
-### global
+Plugins can also run scheduled tasks through the functions:
 
-Global is used for sharing functionality between `setupPlugin` and `processEvent`, since global scope does not work in the context of PostHog plugins. 
+- `runEveryMinute`
+- `runEveryHour`
+- `runEveryDay`
 
-### attachments
+These functions only take an object of type `PluginMeta` as a parameter and do not return anything.
 
-Attachments gives access to files uploaded by the user for config parameters of type `attachment`. An `attachment` has the following type definition:
+Example usage:
 
 ```js
-interface PluginAttachment {
-    content_type: string
-    file_name: string
-    contents: any
+async function runEveryMinute ({cache}) {
+    const cursor = await cache.get('last_fetched_cursor')
+    const response = await fetch(`http://my.api.com/get_stargazers?since=${cursor}`)
+    for (const { username } of response.stargazers) {
+        posthog.capture('someone starred you!', { username })
+    }
+    const last_cursor = response.stargazers.map(s => s.created_at).sort().reverse()[0] || cursor
+    await cache.set('last_fetched_cursor', last_cursor)
 }
 ```
 
-As such, accessing the contents of an uploaded file can be done with `attachments.attachmentName.contents`.
+It's worth noting that the plugin server supports debouncing, meaning that the counter for the next task will only start once the previous task finishes. In other words, if a given task that runs "every minute" takes longer than a minute, the next task will only start one minute after the previous task finishes.
```

Comment:
👍 

Also I'd add a few things:
1) the "every minute" / "every hour" is a promise not a guarantee. A worker may be down for 2 seconds because of a restart and miss the task. We'll add better timing guarantees in the future. 
2) We'll also add better configurations, so "runAt4pm" vs "runEvery15Min", or maybe change the API altogether.
3) If you have multiple instances of posthog-plugin-server running, the defined tasks will be run on each instance at the specified interval. So things might run 2x, 3x, if you have 2, 3, etc dynos running. Fixes for this will come soon as well.